webui/react/src/hooks/useModal/UserSettings/useModalNameChange.test.tsx

```diff
@@ -0,0 +1,146 @@
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { Button } from 'antd';
+import React, { useCallback, useEffect } from 'react';
+
+import StoreProvider, { StoreAction, useStoreDispatch } from 'contexts/Store';
+import { PatchUserParams } from 'services/types';
+import { DetailedUser } from 'types';
+
+import useModalUserSettings from './useModalUserSettings';
+
+const mockPatchUser = jest.fn();
+
+jest.mock('services/api', () => ({
+  patchUser: (params: PatchUserParams) => {
+    return mockPatchUser(params);
+  },
+}));
+
+const OPEN_MODAL_TEXT = 'Open Modal';
+const USERNAME = 'test_username1';
+const USER_ID = 1;
+const DISPLAY_NAME = 'Test Name';
+const NAME_CHANGE_TEXT = 'Change name';
+const USER_SETTINGS_HEADER = 'Account';
+const UPDATED_DISPLAY_NAME = 'New Displayname';
+
+const currentUser: DetailedUser = {
+  displayName: DISPLAY_NAME,
+  id: USER_ID,
+  isActive: true,
+  isAdmin: false,
+  username: USERNAME,
+};
+
+const users: Array<DetailedUser> = [ currentUser ];
+
+const TestApp: React.FC = () => {
+  const { contextHolder, modalOpen: openUserSettingsModal } = useModalUserSettings();
+  const storeDispatch = useStoreDispatch();
+
+  const loadUsers = useCallback(() => {
+    storeDispatch({
+      type: StoreAction.SetUsers,
+      value: users,
+    });
+    storeDispatch({
+      type: StoreAction.SetCurrentUser,
+      value: currentUser,
+    });
+  }, [ storeDispatch ]);
+
+  useEffect(() => {
+    loadUsers();
+  });
+
+  return (
+    <div>
+      {contextHolder}
+      <Button onClick={() => openUserSettingsModal()}>
+        {OPEN_MODAL_TEXT}
+      </Button>
+    </div>
+  );
+};
+
+const setup = async () => {
+  const view = render(
+    <StoreProvider>
+      <TestApp />
+    </StoreProvider>,
+  );
+  const user = userEvent.setup();
+  await user.click(await screen.findByText(OPEN_MODAL_TEXT));
+  await user.click(await screen.findByText(NAME_CHANGE_TEXT));
+
+  return { user, view };
+};
+
+describe('useModalNameChange', () => {
+  it('opens modal with correct values', async () => {
```

Comment:
it would have been valuable if we could get jest to use a different testing engine in our local machines for testing the tests. I've written tests before without having checked that they fail. if you can see tests running (slowed down and in a browser, while you see the description "it should ...." you can can some of those cases.